checkbox-provider-ce-oem/bin/thermal_sensor_test.py

```diff
@@ -0,0 +1,188 @@
+
+    stderr_handler = logging.StreamHandler(sys.stderr)
+    stderr_handler.setFormatter(logging.Formatter(logger_format, date_format))
+
+    stdout_handler.setLevel(logging.DEBUG)
+    stderr_handler.setLevel(logging.WARNING)
+
+    # Add a filter to the stdout handler to limit log records to
+    # INFO level and below
+    stdout_handler.addFilter(lambda record: record.levelno <= logging.INFO)
+
+    root_logger.addHandler(stderr_handler)
+    root_logger.addHandler(stdout_handler)
+
+    return root_logger
+
+
+class ThermalMonitor():
+
+    def __init__(self, name):
+        self._name = name
+        self.root_node = Path(SYS_THERMAL_PATH).joinpath(self._name)
+        self.type_node = self.root_node.joinpath("type")
+        self.temp_node = self.root_node.joinpath("temp")
+        self.mode_node = self.root_node.joinpath("mode")
+        self.initial_temp = None
+
+    def _read_node(self, node):
+        if node.exists():
+            return node.read_text().strip("\n")
+        else:
+            raise FileNotFoundError("{} file not exists".format(str(node)))
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type(self):
+        return self._read_node(self.type_node)
+
+    @property
+    def temperature(self):
+        temp = self._read_node(self.temp_node)
+        if not temp.isnumeric():
+            raise ValueError("temperate value is not a number!")
+        return temp
+
+    @property
+    def mode(self):
+        return self._read_node(self.mode_node)
+
+
+def check_temperature(current, initial):
+    logging.info("Initial value: %s, current value: %s", initial, current)
+    return (int(current) != 0 and current != initial)
+
+
+def thermal_monitor_test(args):
+    logging.info(
+        "# Monitor the temperature of %s thermal around %s seconds",
+        args.name, args.duration)
+
+    if args.extra_commands == "stress-ng":
+        cmd = ("stress-ng --cpu 0 --io 4 --vm 2 "
+               "--vm-bytes 128M --timeout {}s").format(args.duration)
+    else:
+        cmd = args.extra_commands
+
+    thermal_op = ThermalMonitor(args.name)
+    if thermal_op.mode == "disabled":
+        raise SystemExit("Error: The {}-{} thermal is disabled".format(
+                                            thermal_op.name, thermal_op.type))
+    initial_value = thermal_op.temperature
+
+    result = False
+    try:
+        proc = subprocess.Popen(shlex.split(cmd))
+    except Exception:
+        # Bypass any error while issue command through Popen
+        # Due to the command here is trying to increase system loading
+        pass
+
+    for _ in range(args.duration):
+        cur_temp = thermal_op.temperature
+        result = check_temperature(cur_temp, initial_value)
+        if result:
+            logging.info(
+                "# The temperature of %s thermal has been altered",
+                args.name)
+            break
+        time.sleep(1)
+    if proc.poll() is None:
```

Comment:
If the cmd is not existed, the proc will be none, and raise the `UnboundLocalError`.
```python
if proc and proc.poll() is None:
```